docs/scripting-ref.rst

```diff
@@ -391,6 +393,124 @@ Compare:
+  to execute before cancelling it and returning control to Lua. The timeout
+  must be strictly greater than zero and less than or equal to 60. The timeout
+  defaults to 5 seconds.
+
+**Returns:** ``msg, error`` pair. When the execution is successful
+``msg`` is either nil or contains a stringified JavaScript value. In case of
+timeout or JavaScript errors ``msg`` is ``nil`` and ``error`` contains
+an error message string.
+
+Example:
+
+.. code-block:: lua
+
+    local msg, error = splash:wait_for_resume([[
+        function main(splash) {
+            setTimeout(function () {
+                splash.resume("Hello, world!");
+            }, 3000);
+        }
+    ]])
+
+    print(msg, error)
+
+This trivial example shows how to transfer control of execution from Lua
+to JavaScript and then back to Lua. This command will tell JavaScript to
+sleep for 3 seconds and then return to Lua. Note that this is an async
+operation: the Lua event loop and the JavaScript event loop continue to run
+during this 3 second pause, but the Lua ``print()`` function will not be called
+until JavaScript calls ``splash.resume()``.
+
+.. note::
+
+    The ``main()`` function is required, and you will get an error if
+    you do not include it. The first argument to this function can have any
+    name you choose, of course. We will call it ``splash`` by convention in
+    this documentation.
+
+The example prints ``Hello, world!   nil`` on the console. The
+``Hello, world!`` is the string returned by ``splash.resume()`` and the ``nil``
+indicates that no error occurred.
+
+.. code-block:: lua
+
+    local msg, error = splash:wait_for_resume([[
+        function main(splash) {
+            setTimeout(function () {
+                splash.error("Goodbye, cruel world!");
+            }, 3000);
+        }
+    ]])
+
+    print(msg, error)
+
+This example shows a similar flow of execution as the previous example,
+but it uses ``splash.error()`` to return an error value instead of a
+success value. The example prints ``nil     error: Goodbye, cruel world!``.
+The ``nil`` indicates no ``msg`` was returned, and
+``error: Goodbye, cruel world!`` contains the error string returned from
+``splash.error()``.
+
+.. code-block:: lua
+
+    local msg, error = splash:wait_for_resume([[
+        function main(splash) {
+            setTimeout(function () {
+                splash.resume("Hello, world!");
+            }, 3000);
+        }
+    ]], 1)
+
+    print(msg, error)
+
+This example is similar to the first example, but here we have changed the
+``timeout`` argument from its default 5 seconds to just 1 second. Of course,
+our JavaScript code will not call ``splash.resume()`` for 3 seconds, which
+guarantees that it will time out. The example prints
+``nil     error: Timed out while waiting for resume() or error().``
+
+.. note::
+
+    Remember that the timeout t must satisfy ``0 < t <= 60`` seconds. You
```

Comment:
I added that for Scraping Hub's benefit :) In a previous version, I used `timeout==0` to indicate no timeout at all. I was worried if the user's code never calls `resume()` or `error()`, then all the resources associated with that request will never be released, right? Or does Splash have a master request timeout? I don't want to rely on the client's timeout, since they can manipulate that to tie up server resources for a very long time.

Is it okay for the user to opt out of the timeout? Or do you still want the `>0` constraint?